gobblin-cluster/src/main/java/org/apache/gobblin/cluster/GobblinHelixTask.java

```diff
@@ -67,113 +57,63 @@
     try (Closer closer = Closer.create()) {
       closer.register(MDC.putCloseable(ConfigurationKeys.JOB_NAME_KEY, this.jobName));
       closer.register(MDC.putCloseable(ConfigurationKeys.JOB_KEY_KEY, this.jobKey));
-      Path workUnitFilePath =
-          new Path(this.taskConfig.getConfigMap().get(GobblinClusterConfigurationKeys.WORK_UNIT_FILE_PATH));
-
-      String fileName = workUnitFilePath.getName();
-      String storeName = workUnitFilePath.getParent().getName();
-      WorkUnit workUnit;
-
-      if (workUnitFilePath.getName().endsWith(AbstractJobLauncher.MULTI_WORK_UNIT_FILE_EXTENSION)) {
-        workUnit = stateStores.mwuStateStore.getAll(storeName, fileName).get(0);
-      } else {
-        workUnit = stateStores.wuStateStore.getAll(storeName, fileName).get(0);
-      }
-
-      // The list of individual WorkUnits (flattened) to run
-      List<WorkUnit> workUnits = Lists.newArrayList();
-
-      if (workUnit instanceof MultiWorkUnit) {
-        // Flatten the MultiWorkUnit so the job configuration properties can be added to each individual WorkUnits
-        List<WorkUnit> flattenedWorkUnits =
-            JobLauncherUtils.flattenWorkUnits(((MultiWorkUnit) workUnit).getWorkUnits());
-        workUnits.addAll(flattenedWorkUnits);
-      } else {
-        workUnits.add(workUnit);
-      }
-
-      globalBroker = SharedResourcesBrokerFactory.createDefaultTopLevelBroker(
-          ConfigFactory.parseProperties(this.jobState.getProperties()), GobblinScopeTypes.GLOBAL.defaultScopeInstance());
-      SharedResourcesBroker<GobblinScopeTypes> jobBroker =
-          globalBroker.newSubscopedBuilder(new JobScopeInstance(this.jobState.getJobName(), this.jobState.getJobId())).build();
-
-      this.taskAttempt = this.taskAttemptBuilder.build(workUnits.iterator(), this.jobId, this.jobState, jobBroker);
-      this.taskAttempt.runAndOptionallyCommitTaskAttempt(GobblinMultiTaskAttempt.CommitPolicy.IMMEDIATE);
-      return new TaskResult(TaskResult.Status.COMPLETED, String.format("completed tasks: %d", workUnits.size()));
+      int workUnitSize = this.task.run();
+      return new TaskResult(TaskResult.Status.COMPLETED, String.format("completed tasks: %d", workUnitSize));
     } catch (InterruptedException ie) {
       Thread.currentThread().interrupt();
       return new TaskResult(TaskResult.Status.CANCELED, "");
     } catch (Throwable t) {
-      LOGGER.error("GobblinHelixTask failed due to " + t.getMessage(), t);
-      return new TaskResult(TaskResult.Status.ERROR, Throwables.getStackTraceAsString(t));
-    } finally {
-      if (globalBroker != null) {
-        try {
-          globalBroker.close();
-        } catch (IOException ioe) {
-          LOGGER.error("Could not close shared resources broker.", ioe);
-        }
-      }
+      _logger.error("GobblinHelixTask failed due to " + t.getMessage(), t);
+      return new TaskResult(TaskResult.Status.FAILED, Throwables.getStackTraceAsString(t));
```

Comment:
How come the state is changed to FAILED? ERROR means helix may retry.

Reply:
ERROR has been deprecated and replaced with FAILED.
see the Helix doc for details.